### PR TITLE
feat(CategoryChannel): add createChannel shortcut method

### PR DESCRIPTION
**Pull request by ImRodry**
**Please describe the changes this PR makes and why it should be merged:**
This PR adds a method to CategoryChannel called createChannel that creates a channel inside that category. It also fixes a bug in the options interface to disallow GUILD_NEWS_THREAD types on both the GuildChannelCreateOptions and the CategoryCreateChannelOptions interfaces

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: src/structures/CategoryChannel.js
Comment (NotSugden) [inline src/structures/CategoryChannel.js:55]: i think `...options` should be done first, just incase a parent is passed in the options by accident or however

File: typings/index.d.ts
Comment (NotSugden) [inline typings/index.d.ts:3229]: maybe out of acope for this PR, but i think this should have a better type

```ta
type ExcludeEnum<Enum, Keys extends keyof Enum> = Exclude<keyof typeof Enum | Enum, Keys | Enum[Keys]>
```

because as-is this type is unnessercerily big

File: src/structures/CategoryChannel.js
Comment (ImRodry) [reply src/structures/CategoryChannel.js:55]: Good point, GitHub copilot suggested this and I didn't even think twice lol

File: typings/index.d.ts
Comment (ImRodry) [reply typings/index.d.ts:3229]: I'm not experienced with types on this level so if you could provide an example of how this would look like I could add it, and since the PR already touches this type I don't see why not

Comment (NotSugden) [reply typings/index.d.ts:3229]:
```ts
ExcludeEnum<ChannelType,
  'DM'
>
```
instead of
```ts
Exclude<keyof typeof ChannelType | ChannelType,
  'DM',
  ChannelType.DM
>
```

Comment (ImRodry) [reply typings/index.d.ts:3229]: Will try this and commit if it goes well

Comment (NotSugden) [reply typings/index.d.ts:3229]: the type might need a bit of fine tuning but yea

File: src/structures/CategoryChannel.js
Comment (vladfrangu) [inline src/structures/CategoryChannel.js:52]: Maybe don't default options

And before you say, yes, it'll be fine
image

**Review by vladfrangu — APPROVED** [review]
Smallest nit, but can be merged without it so LGTM

File: src/structures/CategoryChannel.js
Comment (NotSugden) [reply src/structures/CategoryChannel.js:52]: i think there are a few places where
`{ ...(x ?? {}) }` is used perhaps that could be changed if undefined is valid

Comment (ImRodry) [reply src/structures/CategoryChannel.js:52]: Never seen that notation Sugden but I’ll change it since the main method already defaults these to an empty object

**Comment by anandre** [comment]
I don't think that this method brings any needed utility to the library.  Adding a method that AFAIK is only on managers to separate class seems like it's cluttering up the library for no good reason.  All this does is add more to the library for the convenience of setting one value for the user.  Recently, the library has been removing extraneous helper/utility methods, and this is going in exactly the opposite direction.

File: typings/index.d.ts
Comment (ImRodry) [reply typings/index.d.ts:3229]: Decided I'll add this in a different PR after this one is merged since I also wanna use this ExcludeEnum type in other places, so I'll mark this as resolved for now

**Comment by ImRodry** [comment]
@anandre the methods the library has been recently removing are only ones that were unreliable and caused more hassles than they solved (e.g. typing-related properties and methods, User#presence, etc). This method is indeed a convenience and I recognize that it would be better to have a children manager, however, that cannot be done until v14. If the reviewers prefer we can hold this off and transform CategoryChannel#children into a manager later on, however I don't see why this couldn't be added personally as it is quite convenient and seems to be getting some positive feedback too.

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**
